core/src/scheme/scheme.rs

```diff
@@ -137,6 +154,11 @@ impl Scheme {
         db.contains(&self.state_root())
     }
 
+    pub fn chain_initialized(&mut self, validators: CompactValidatorSet, consensus_params: ConsensusParams) {
```

Comment:
I'm not sure about this change. IMO, if the genesis validators and consensus params are not described in the scheme file, `Scheme` should not have them.

Reply:
I think you need to remove genesis_consensus_params and genesis_validators from Scheme.

Reply:
I put `genesis_consensus_params` and `genesis_validators` in Scheme because I wanted to initialize state at once, by `Scheme::initialize_state`.
I agree that it's a bad idea to put the non-scheme fields to Scheme.

I'm looking for the best place to update `genesis_consensus_params` and `genesis_validators`.
Do you have any idea on this?